cpp/ql/src/semmle/code/cpp/exprs/BuiltInOperations.qll

```diff
@@ -504,15 +894,30 @@ class BuiltInOperationIsFinal extends BuiltInOperation, @isfinalexpr {
 }
 
 /**
- * The `__builtin_choose_expr` type trait.
+ * The `__builtin_choose_expr` expression.  This is a GNU/Clang extension.
+ *
+ * The expression functions similarly to the ternary `?:` operator, except
+ * that it is evaluated at compile-time.
+ * ```
+ * int sz = __builtin_choose_expr(__builtin_types_compatible_p(int, long), 4, 8);
+ * ```
  */
 class BuiltInChooseExpr extends BuiltInOperation, @builtinchooseexpr {
   override string toString() { result = "__builtin_choose_expr" }
+
+  override string getCanonicalQLClass() { result = "BuiltInChooseExpr" }
 }
 
 /**
- * Fill operation on a GNU vector.
+ * Fill operation on a vector. This is a GNU extension.
+ * ```
+ * typedef float float4 __attribute__((ext_vector_type(4)));
+ * float4 v4si = (float4){ 1.0, 2.0, 3.0, 4.0 };
```

Comment:
Which line contains the fill operation?

Reply:
I was hoping this to be the second line but, truth be told, I'm having trouble triggering `VectorFillOperation` with the above C constructs (and some variations thereof).  Since `VectorFillOperation extends @vec_fill`, perhaps the extractor is to blame here?  I think, for the time being, I'll scrap the example altogether,

Reply:
They're not very common, but there are some in [qemu/qemu](https://lgtm.com/projects/g/qemu/qemu/snapshot/36f68942807f5886c4a2ba46177eeccf7d176dc8/files/accel/tcg/tcg-runtime-gvec.c?sort=name&dir=ASC&mode=heatmap#L771).  Or you can remove the example for now and we can perhaps come back to it later?

Reply:
Thanks for pointing me towards `qemu/quemu', I'll look up the syntax and usage therein, and hopefully we'll come up with some sort of workable example.  Finding corresponding concrete syntax for many QL classes is tricky...

Reply:
Thanks for addressing this.